File: src/main/scala/arriba/scala/fix/session/AlwaysResolvingSessionResolver.scala
package arriba.scala.fix.session

import collection.mutable.HashMap
import arriba.scala.fix.messages.FixMessage
import arriba.scala.common.{MapHandlerRepository, PrintingHandler, Handler}

class AlwaysResolvingSessionResolver extends SessionResolver {

  val messageTypeToHandler = new HashMap[String, Handler[FixMessage]]
  messageTypeToHandler.put("D", new PrintingHandler[FixMessage])
  val session = new Session(new SimpleSessionId("targetCompId"),
    new MapHandlerRepository[String, FixMessage](messageTypeToHandler))

  def resolve(sessionId: SessionId): Session = {
    session
  }
}

File: src/main/java/arriba/fix/session/AlwaysResolvingSessionResolver.java
package arriba.fix.session;

import java.util.HashMap;
import java.util.Map;

import arriba.common.Handler;
import arriba.common.MapHandlerRepository;
import arriba.common.PrintingHandler;
import arriba.fix.inbound.FixMessage;

public final class AlwaysResolvingSessionResolver implements SessionResolver {

    private final Session session;

    public AlwaysResolvingSessionResolver() {
        final Map<String, Handler<FixMessage>> messageTypeToHandler = new HashMap<String, Handler<FixMessage>>();
        messageTypeToHandler.put("D", new PrintingHandler<FixMessage>());

        this.session = new Session(new SimpleSessionId("targetCompId"),
                new MapHandlerRepository<String, FixMessage>(messageTypeToHandler));
    }

    public Session resolve(final SessionId sessionId) throws UnknownSessionIdException {
        return this.session;
    }

}
